backend/orchestrator/resources.py

```diff
@@ -125,11 +125,21 @@ def from_grpc(cls, i: algo_pb2.AlgoInput) -> "AlgoInput":
         return cls(kind=AssetKind.from_grpc(i.kind), multiple=i.multiple, optional=i.optional)
 
 
+class AlgoOutput(pydantic.BaseModel):
+    kind: AssetKind
+    multiple: bool
+
+    @classmethod
+    def from_grpc(cls, o: algo_pb2.AlgoOutput) -> "AlgoOutput":
```

Comment:
Since python 3.7+, `from __future__ import annotations` can be used to return `AlgoOutput` class as type annotation instead of the string `"AlgoOutput"`. As we don't want have compatibility with python 3.6, it can be a nice improvement.
A good explanation can be found [here](https://stackoverflow.com/a/33533514)
It can be done in another pull request.

Reply:
Good point, I'll do it here. This will all be replaced by `Self` once we switch to 3.11 🎉